### PR TITLE
Add ROADMAP.md to share our quarterly plans publicly.

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
@matt-gardner what detail would you add to the efforts to split out models into their own library?

@brendan-ai2 could you link to the particular issues behind each performance improvement?

File: ROADMAP.md [hunk 1]
```diff
@@ -12,14 +12,19 @@ We have a few major themes for this quarter:
...
     a.  [Semantic Parsing](https://github.com/allenai/allennlp-semparse)
+    
     b.  [Language Modeling](https://github.com/allenai/allennlp-language-modeling)
 
 2.  Better supporting our initiative around [Green AI](https://arxiv.org/abs/1907.10597).
 
     a.  Improve the performance of AllenNLP.  In particular, see the following issues:
+    
         i.
+
```
Comment (brendan-ai2) [inline ROADMAP.md]: "Using DistributedDataParallel for multi GPU training" https://github.com/allenai/allennlp/issues/2536

File: ROADMAP.md [hunk 2]
```diff
@@ -12,14 +12,19 @@ We have a few major themes for this quarter:
...
     b.  [Language Modeling](https://github.com/allenai/allennlp-language-modeling)
 
 2.  Better supporting our initiative around [Green AI](https://arxiv.org/abs/1907.10597).
 
     a.  Improve the performance of AllenNLP.  In particular, see the following issues:
+    
         i.
+        
         ii.
+
```
Comment (brendan-ai2) [inline ROADMAP.md]: "Native pytorch Multiprocessing for data loading" https://github.com/allenai/allennlp/issues/3079

File: ROADMAP.md [hunk 3]
```diff
@@ -12,14 +12,19 @@ We have a few major themes for this quarter:
...
 2.  Better supporting our initiative around [Green AI](https://arxiv.org/abs/1907.10597).
 
     a.  Improve the performance of AllenNLP.  In particular, see the following issues:
+    
         i.
+        
         ii.
+        
         iii.
+
```
Comment (brendan-ai2) [inline ROADMAP.md]: "Mixed Precision Training" https://github.com/allenai/allennlp/issues/2149

File: ROADMAP.md [hunk 4]
```diff
@@ -12,14 +12,19 @@ We have a few major themes for this quarter:
...
     a.  Improve the performance of AllenNLP.  In particular, see the following issues:
+    
         i.
+        
         ii.
+        
         iii.
+        
     b.  Add tooling so any user can easily measure the total number of floating-point operations
     performed during training.
```
Comment (brendan-ai2) [inline ROADMAP.md]: Current tracking ticket is here: https://github.com/allenai/allennlp/issues/3436

I should note that this roadmap entry is more aggressive than the goal as currently written. I think that's okay in that the roadmap version is certainly where we want to end up, but it's a bit further out.

**Comment by brendan-ai2** [comment]
@schmmd, I added some links inline! Let me know if you'd like me to elaborate on anything. Also, Matt will naturally be able to give further context, but one issue for the splitting work is https://github.com/allenai/allennlp/issues/3351.

**Comment by schmmd** [comment]
@matt-gardne ping: what detail would you add to the efforts to split out models into their own library?

File: ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+We have a few major themes for this quarter:
+
+1.  Breaking out models into their own repositories.  By doing so we hope to increase community
+    engagement and ownership of these model repositories, allowing us to focus more on the core
+    library.  We already have a good start on this with the following repositories:
+
+    a.  [Semantic Parsing](https://github.com/allenai/allennlp-semparse)
+    
+    b.  [Language Modeling](https://github.com/allenai/allennlp-language-modeling)
```
Comment (matt-gardner) [inline ROADMAP.md]: I would list reading comprehension here instead, it's a lot closer to ready.  The LM one is just a skeleton so far.  I would add this after these lines:

```
We will test out this idea of splitting up the repositories with these two first,
fixing any issues with them by the end of this quarter.  Around the end of
this quarter or early next quarter, we will finish the process with the rest of
the models, making repositories around language modeling, coreference
resolution, sequence tagging, parsing, etc.
```

Comment (matt-gardner) [reply ROADMAP.md]: And you can add a reference to the ticket that @brendan-ai2 mentioned.

**Pull request merged**